pkg/koordlet/metricsadvisor/collectors/performance/performance_collector_linux.go

```diff
@@ -227,24 +238,42 @@ func (p *performanceCollector) collectSingleContainerPSI(podParentCgroupDir stri
 		klog.Errorf("collect container %s psi err: %v", containerStatus.Name, err)
 		return
 	}
-	containerPsiMetric := &metriccache.ContainerInterferenceMetric{
-		MetricName:  metriccache.MetricNameContainerPSI,
-		PodUID:      string(pod.UID),
-		ContainerID: containerStatus.ContainerID,
-		MetricValue: &metriccache.PSIMetric{
-			SomeCPUAvg10:     containerPSI.CPU.Some.Avg10,
-			SomeMemAvg10:     containerPSI.Mem.Some.Avg10,
-			SomeIOAvg10:      containerPSI.IO.Some.Avg10,
-			FullCPUAvg10:     containerPSI.CPU.Full.Avg10,
-			FullMemAvg10:     containerPSI.Mem.Full.Avg10,
-			FullIOAvg10:      containerPSI.IO.Full.Avg10,
-			CPUFullSupported: containerPSI.CPU.FullSupported,
-		},
+
+	cpuSomeAvg10, err01 := metriccache.ContainerPSISomeCPUAvg10Metric.GenerateSample(
```

Comment:
we can implement as for loops here if we just define ContainerPSI as the metric.
for cpu, mem, io:
  for some, full:
    metric := generateSample
    metricList = append(metricList, metric)
return metricList